collect_app/src/main/java/org/odk/collect/android/preferences/AutoSendPreferenceMigrator.java

```diff
@@ -0,0 +1,83 @@
+package org.odk.collect.android.preferences;
+
+import android.annotation.SuppressLint;
+import android.content.SharedPreferences;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Map;
+
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_AUTOSEND;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_AUTOSEND_NETWORK;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_AUTOSEND_WIFI;
+
+/**
+ * Migrates existing autosend_wifi and autosend_network preference values to autosend
+ */
+public class AutoSendPreferenceMigrator {
+
+    private static boolean autoSendWifi = false;
+    private static boolean autoSendNetwork = false;
+    private static String autoSend = "off";
+
+    @SuppressLint("ApplySharedPref")
```

Comment:
https://github.com/opendatakit/collect/pull/653
`apply` saves the data asynchronously. Thus it doesn't block the UI thread and is faster than `commit`.